workitem/workitem_repository.go

```diff
@@ -600,7 +600,33 @@ func (r *GormWorkItemRepository) Create(ctx context.Context, spaceID uuid.UUID,
 	if !wiType.CanConstruct {
 		return nil, errors.NewForbiddenError(fmt.Sprintf("cannot construct work items from \"%s\" (%s)", wiType.Name, wiType.ID))
 	}
-
+	var allowedWorkitemTypesID []uuid.UUID
+
+	// Prohibit creation of work items from a type that doesn't belong to current space template
+	query := fmt.Sprintf(`
+		SELECT id from %s where space_template_id = (
+			SELECT space_template_id from spaces where id::text = '%s'
+		)`, wiType.TableName(), spaceID.String())
+	db := r.db.Raw(query)
+	db.Pluck("id", &allowedWorkitemTypesID)
```

Comment:
I think we can make this much simpler by checking if the work item type (`typeID`) exists that has the same template ID as the one of the space `spaceID`. So something like this:

```sql
SELECT EXISTS ( 
  SELECT 1 FROM work_item_types WHERE id = $1 AND space_template_id = (
    SELECT space_template_id FROM spaces WHERE id = $2
  )
)
-- $1 = typeID
-- $2 = spaceID
```

Please substitute the table names as you did.

If that row exists, then we can create the work item because it uses a work item type that belongs to the correct space template.

Reply:
Thank you @kwk. This is much better than what I had implemented.

Please see https://github.com/fabric8-services/fabric8-wit/pull/2115/commits/dedbb165140bfa43efcafa98c3908a4c766856c1#diff-55f2da878ed2fcb42e5373010f02ff92R606